### ISSUE TITLE
DeDRM doesn't work from Adobe Digital Editions

### ISSUE THREAD

**Issue by schmay02** [apprenticeharper/DeDRM_tools#278]
calibre Fehlerdiagnoseprotokoll
calibre 2.85.1  embedded-python: True is64bit: False
Windows-8-6.2.9200 Windows ('32bit', 'WindowsPE')
32bit process running on 64bit windows
('Windows', '8', '6.2.9200')
Python 2.7.9
Windows: ('8', '6.2.9200', '', 'Multiprocessor Free')
Successfully initialized third party plugins: DeDRM (6, 5, 3) && EpubSplit (2, 2, 0) && Obok DeDRM (6, 3, 6)
devicePixelRatio: 1.0
logicalDpi: 192.0 x 192.0
physicalDpi: 277.406143345 x 277.090909091
Starting up...
DEBUG:    0.0 obok::utilities.py - loading translations
DEBUG:    0.0 obok::dialogs.py - loading translations
DEBUG:    0.0 obok::config.py - loading translations
DEBUG:    0.0 obok::action_err.py - loading translations
Started up in 4.99 seconds with 60 books
Storage number map:
{(7L, 0L): [(3L, 'C'), (4L, 'D')], (7L, 1L): [(1L, 'E')]}
Storage number for USBSTOR\DISK&VEN_KINDLE&PROD_INTERNAL_STORAGE&REV_0100\90231702251508NC&0: StorageDeviceNumber(type=7L, number=1L, partition_number=0L)
Drive letters for USBDevice(vendor_id=0x1949 product_id=0x4 bcd=0x100 devid=usb\vid_1949&pid_0004&rev_0100 devinst=6)
{u'drive_letters': ['E'],
 u'pnp_id_map': {'E': u'\\\\?\\usbstor#disk&ven_kindle&prod_internal_storage&rev_0100#90231702251508nc&0#{53f56307-b6bf-11d0-94f2-00a0c91efb8b}'},
 u'readonly_drives': set([]),
 u'sort_map': {'E': (1L, 1L)}}
Job: 1 Geräteinformationen erhalten started
DeviceJob: 1 Geräteinformationen erhalten done, calling callback
DeviceJob: 1 Geräteinformationen erhalten callback returned
Job: 2 Bibliotheksinformationen festlegen started
DeviceJob: 2 Bibliotheksinformationen festlegen done, calling callback
DeviceJob: 2 Bibliotheksinformationen festlegen callback returned
Job: 3 Liste der Bücher auf dem Gerät erhalten started
DEBUG:    0.0 USBMS: Fetching list of books from device. Device= KINDLE2 oncard= None
DEBUG:    0.0 USBMS: dirs are: E:\ documents
DEBUG:    0.0 USBMS: scan from root False E:\documents
DEBUG:    0.1 USBMS: count found in cache: 61, count of files in metadata: 61, need_sync: False
DEBUG:    0.1 USBMS: Finished fetching list of books from device. oncard= None
DEBUG:    0.1 USBMS: Fetching list of books from device. Device= KINDLE2 oncard= carda
DEBUG:    0.1 USBMS: Fetching list of books from device. Device= KINDLE2 oncard= cardb
DeviceJob: 3 Liste der Bücher auf dem Gerät erhalten done, calling callback
DeviceJob: metadata_downloaded: Starting set_books_in_library
DeviceJob: set_books_in_library: books to process= 61
DeviceJob: set_books_in_library finished: time= 0.00399994850159
DeviceJob: metadata_downloaded: updating views
DeviceJob: metadata_downloaded: syncing
DeviceJob: metadata_downloaded: refreshing ondevice
DeviceJob: metadata_downloaded: sending metadata_available signal
DeviceJob: 3 Liste der Bücher auf dem Gerät erhalten callback returned
Job: 2 Bibliotheksinformationen festlegen finished
Keine weiteren Informationen verfügbar.
Job: 1 Geräteinformationen erhalten finished
Keine weiteren Informationen verfügbar.
Job: 3 Liste der Bücher auf dem Gerät erhalten finished
Keine weiteren Informationen verfügbar.
stdout+stderr from file dialog helper: ['', '']
piped data from file dialog helper: ['\x07\x1e\x07*QK~<FI\xa9\x9dKH\xb6\xf2`Qaw\xc0\xfe\xf0R \x97\xac=\x08\xec\x8e\x8a', 'C:\\Users\\lemon\\OneDrive\\Dokumente\\My Digital Editions\\Archer J.,Erbe und Schicksal (Clifton 3)-2.epub']
DeDRM v6.5.3: Trying to decrypt Archer J.,Erbe und Schicksal (Clifton 3)-2.epub
DeDRM v6.5.3: Verifying zip archive integrity
DeDRM v6.5.3: Archer J.,Erbe und Schicksal (Clifton 3)-2.epub is a secure Adobe Adept ePub
DeDRM v6.5.3: Looking for new default Adobe Digital Editions Keys after 0.2 seconds
DeDRM v6.5.3: Exception when getting default Adobe Key after 0.2 seconds
Traceback (most recent call last):
  File "calibre_plugins.dedrm.__init__", line 337, in ePubDecrypt
  File "calibre_plugins.dedrm.adobekey", line 378, in adeptkeys
  File "calibre_plugins.dedrm.adobekey", line 358, in CryptUnprotectData
ADEPTError: Failed to decrypt user key key (sic)
DeDRM v6.5.3: Ultimately failed to decrypt after 0.2 seconds.
Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Running file type plugin DeDRM failed with traceback:
Traceback (most recent call last):
  File "site-packages\calibre\customize\ui.py", line 170, in _run_filetype_plugins
  File "calibre_plugins.dedrm.__init__", line 621, in run
  File "calibre_plugins.dedrm.__init__", line 392, in ePubDecrypt
DeDRMError: DeDRM v6.5.3: Ultimately failed to decrypt after 0.2 seconds.
Added Archer J.,Erbe und Schicksal (Clifton 3) to db in: 0.2
Added 1 books in 1.1 seconds
Exception TypeError: "'NoneType' object is not callable" in <bound method NativeFunction.__del__ of <calibre_plugins.dedrm.adobekey.NativeFunction object at 0x04DD32B0>> ignored
Exception TypeError: "'NoneType' object is not callable" in <bound method NativeFunction.__del__ of <calibre_plugins.dedrm.adobekey.NativeFunction object at 0x04DD3250>> ignored

**Comment by ElleKayEm** [comment]
Can you read the book in ADE 2.0.1?  Authorized with your Adobe ID?  On the same machine and user as calibre?

**Comment by schmay02** [comment]
Yes, I can read the ebook in ADE 2.0.1, authorized with my ID, but still can't convert books in Calibre using the same laptop and same account. Do I need to use a certain version of Calibre?

**Comment by schmay02** [comment]
calibre Debug log
calibre 2.85.1 [64bit]  embedded-python: True is64bit: True
Windows-8-6.2.9200 Windows ('64bit', 'WindowsPE')
('Windows', '8', '6.2.9200')
Python 2.7.9
Windows: ('8', '6.2.9200', '', 'Multiprocessor Free')
Successfully initialized third party plugins: DeDRM (6, 5, 3) && EpubMerge (2, 1, 0) && Obok DeDRM (6, 3, 6) && EpubSplit (2, 2, 0)
devicePixelRatio: 1.0
logicalDpi: 192.0 x 192.0
physicalDpi: 277.406143345 x 277.090909091
Starting up...
DEBUG:    0.0 obok::utilities.py - loading translations
DEBUG:    0.0 obok::dialogs.py - loading translations
DEBUG:    0.0 obok::config.py - loading translations
DEBUG:    0.0 obok::action_err.py - loading translations
EpubMerge: DEBUG: 2017-05-25 14:00:51,272: calibre_plugins.epubmerge.epubmerge_plugin(152): macmenuhack file_path:C:\Users\lemon\AppData\Roaming\calibre\plugins\fanficfare_macmenuhack.txt
Started up in 7.31 seconds with 59 books
stdout+stderr from file dialog helper: ['', '']
piped data from file dialog helper: ['M\xc9\xbb\xadm\xc2!\xdc\x1b\tDj*t\xc7\x1d\x86\xfd\\\xe1\xd0]\xed5\xf7\xdf\x12\x14k\x04m\x8c', 'C:\\Users\\lemon\\OneDrive\\Dokumente\\My Digital Editions\\Archer J.,Erbe und Schicksal (Clifton 3).epub']
DeDRM v6.5.3: Trying to decrypt Archer J.,Erbe und Schicksal (Clifton 3).epub
DeDRM v6.5.3: Verifying zip archive integrity
DeDRM v6.5.3: Archer J.,Erbe und Schicksal (Clifton 3).epub is a secure Adobe Adept ePub
DeDRM v6.5.3: Looking for new default Adobe Digital Editions Keys after 0.1 seconds
DeDRM v6.5.3: Exception when getting default Adobe Key after 0.2 seconds
Traceback (most recent call last):
  File "calibre_plugins.dedrm.__init__", line 337, in ePubDecrypt
  File "calibre_plugins.dedrm.adobekey", line 378, in adeptkeys
  File "calibre_plugins.dedrm.adobekey", line 358, in CryptUnprotectData
ADEPTError: Failed to decrypt user key key (sic)
DeDRM v6.5.3: Ultimately failed to decrypt after 0.2 seconds.
Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Running file type plugin DeDRM failed with traceback:
Traceback (most recent call last):
  File "site-packages\calibre\customize\ui.py", line 170, in _run_filetype_plugins
  File "calibre_plugins.dedrm.__init__", line 621, in run
  File "calibre_plugins.dedrm.__init__", line 392, in ePubDecrypt
DeDRMError: DeDRM v6.5.3: Ultimately failed to decrypt after 0.2 seconds.
Added Archer J.,Erbe und Schicksal (Clifton 3) to db in: 0.3
Added 1 books in 1.1 seconds


I've changed the language to English and created another log file. I hope that helps you.
